.github/workflows/cache.yml

```diff
@@ -0,0 +1,36 @@
+name: Cache Keyman API
+
+on:
+  schedule:
+    - cron: '0 0 * * *'
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Fetch from Keyman API
+        run: |
+          curl https://api.keyman.com/search?q= > src/app/output.json
+
+      - name: Commit JSON output
+        id: commit
+        run: |
+          git config --local user.email "action@github.com"
+          git config --local user.name "github-actions"
+          git add --all
+          if [-z "$(git status --porcelain)"]; then
+            echo "::set-output name=push::false"
+          else
+            git commit -m "Update Keyman API" -a
+            echo "::set-output name=push::true"
+          fi
+        shell: bash
+      
+      - name: Push changes
+        if: steps.commit.outputs.push == 'true'
+        uses: ad-m/github-push-action@master
+        with:
+          github_token: ${{ secrets.GITHUB_TOKEN }}
```

Comment:
It might be easier to use [this action](https://github.com/stefanzweifel/git-auto-commit-action) instead of the current two build steps.